crates/primitives/src/basefee.rs

```diff
@@ -23,42 +23,15 @@
 /// The calculated base fee for the next block as a `u64`.
 ///
 /// For more information, refer to the [EIP-1559 spec](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md).
+use alloy_eips::eip1559::calc_next_block_base_fee;
+
 pub fn calculate_next_block_base_fee(
     gas_used: u64,
     gas_limit: u64,
     base_fee: u64,
     base_fee_params: crate::BaseFeeParams,
 ) -> u64 {
-    // Calculate the target gas by dividing the gas limit by the elasticity multiplier.
-    let gas_target = gas_limit / base_fee_params.elasticity_multiplier as u64;
-
-    match gas_used.cmp(&gas_target) {
-        // If the gas used in the current block is equal to the gas target, the base fee remains the
-        // same (no increase).
-        std::cmp::Ordering::Equal => base_fee,
-        // If the gas used in the current block is greater than the gas target, calculate a new
-        // increased base fee.
-        std::cmp::Ordering::Greater => {
-            // Calculate the increase in base fee based on the formula defined by EIP-1559.
-            base_fee +
-                (std::cmp::max(
-                    // Ensure a minimum increase of 1.
-                    1,
-                    base_fee as u128 * (gas_used - gas_target) as u128 /
-                        (gas_target as u128 * base_fee_params.max_change_denominator),
-                ) as u64)
-        }
-        // If the gas used in the current block is less than the gas target, calculate a new
-        // decreased base fee.
-        std::cmp::Ordering::Less => {
-            // Calculate the decrease in base fee based on the formula defined by EIP-1559.
-            base_fee.saturating_sub(
-                (base_fee as u128 * (gas_target - gas_used) as u128 /
-                    (gas_target as u128 * base_fee_params.max_change_denominator))
-                    as u64,
-            )
-        }
-    }
+    calc_next_block_base_fee(gas_used as u128, gas_limit as u128, base_fee as u128, base_fee_params) as u64
```

Comment:
i'd just re-export this and replace every call in the codebase with `calc_next_block_base_fee`, or alternatively import the alloy_eips crate where this is called

Reply:
I thought of doing that but felt that changing the function name throughout the whole codebase wouldn't be okay for me to do. If there is no issue with my approach can the PR be merged or should I do it again your way?

Reply:
> re-export would be preferred here, even if this means we need to about a few type casts etc

working on it